upload the zsync file

Fuseteam opened the pull request:
i might still need to look into deleting the old zsync file just like it's deleting the old appimage file but otherwise this is ready

Fuseteam commented:
i don't quite see why it failed

Thatoo commented:
It fails because devs didn't yet review it which is the first step.

@FlexW and @mgallien could you review this PR please that would allow to upload the zsync file missing for the AppImage to be able to auto-update .

mgallien commented:
@Fuseteam very sorry for the extra delay on our side to get a proper review

thanks for your work

Thatoo commented:
> no worries, i look forward to the review~
> 
> p.s. i see the DCO is now failing, anything i need to do?

Yes, you'll have to merge the master branch into your Fuseteam:fix-appimage-update branch. However, don't do it right now as you'll have to do it everytime something change in master till you merge so wait for the review and when the review is ok, you merge master into your branch and then DCO will start again and will be, hopefully, OK.

mgallien commented:
> no worries, i look forward to the review~
> 
> p.s. i see the DCO is now failing, anything i need to do?

The best is to follow the instructions [here](https://github.com/nextcloud/desktop/pull/4529/checks?check_run_id=7107615546)

Fuseteam commented:
@mgallien is the review ok?

Fuseteam commented:
@mgallien the DCO is fine now

Fuseteam commented:
bah the rebase broke the DCO xd

Thatoo commented:
@mgallien could you review please and tell @Fuseteam if something is wrong in his code?

Fuseteam commented:
@mgallien are there are blockers left for this PR?

mgallien commented:
> @mgallien are there are blockers left for this PR?

not sure how to test the upgrade path of an AppImage
@Fuseteam do you have pointers to documentation about that ?
despite having to work with AppImage, I am still lacking some good docs reference

mgallien commented:
> @mgallien are there are blockers left for this PR?

DCO is still unhappy
see error message
```
Commit sha: [22bbfef](https://github.com/nextcloud/desktop/pull/4529/commits/22bbfef50d500e4d0a7e885b098144bb41bbeaf2), Author: Rahammetoela Toekiman, Committer: Matthieu Gallien; Expected "Rahammetoela Toekiman [fusekai@outlook.com](mailto:fusekai@outlook.com)", but got "Fuseteam [fusekai@outlook.com](mailto:fusekai@outlook.com)".
```

Fuseteam commented:
> > @mgallien are there are blockers left for this PR?
> 
> not sure how to test the upgrade path of an AppImage @Fuseteam do you have pointers to documentation about that ? despite having to work with AppImage, I am still lacking some good docs reference

i do not, but iirc https://github.com/AppImageCommunity/AppImageUpdate uses the zsync file to detect if there's an update and download it

Fuseteam commented:
@mgallien DCO seems happy now, not sure about the rest

Fuseteam commented:
Oh wait now i see the full error

Fuseteam commented:
@mgallien _now_ it should be fine

Fuseteam commented:
lol 2 years and counting xD

mgallien commented:
@Fuseteam sorry for the extra long delay
I could never really find time to work on this
sorry to ask again (unless I missed if)
what would be still needed to have automated update for AppImage ?
let me then keep an active eye on this to ensure we eventually get all of this done

mgallien commented:
given that the deployment on some proper place to distribute update packages would happen when we do a release similar to what we need to do with sparkle framework on macOS
we can look into modifying our release steps to take this into account (no matter if that is automated or not)
see for example https://github.com/nextcloud/desktop/issues/2599

Fuseteam commented:
> @Fuseteam sorry for the extra long delay I could never really find time to work on this sorry to ask again (unless I missed if) what would be still needed to have automated update for AppImage ? let me then keep an active eye on this to ensure we eventually get all of this done

i think this is supposed to fix it, i believe all we need is to get the `zsync` file uploaded so that the appimage is able to download just the delta on its own

Fuseteam commented:
granted i have moved on to use the snap version as that has automatic updates out of the box. I moved on in part because appimages rely on fuse2 library which has been unmaintained for years at this point (fuse3 is already out) and some distros have begun not shipping fuse2 out of the box. the snap is community maintained, but if you guys would like to incorporate it in the release steps that would be nice too.

That said i'm willing to see this pr to the end

mgallien commented:
@Fuseteam we will need to find another solution as we removed the file modified in the PR when moving from drone CI to gtihub CI
sorry for that and many thanks for the contribution

Fuseteam commented:
well that's fun any tips @Thatoo?

Thatoo commented:
no, sorry

mgallien commented:
as I said, very sorry that we missed the connection between this PR and the move of AppImage CI from drone to github actions
for releases, if I understand it correctly, we would *just* need to produce the zsync file and upload it to the release artifacts
then I have to figure out how we run the upgrade itself (this is what I never managed to understand/learn)

Fuseteam commented:
> as I said, very sorry that we missed the connection between this PR and the move of AppImage CI from drone to github actions for releases, if I understand it correctly, we would _just_ need to produce the zsync file and upload it to the release artifacts then I have to figure out how we run the upgrade itself (this is what I never managed to understand/learn)

tis a shame, but yes. the appimage updater looks for zsync file in order to resolve differences as far as i remember.
